cylc/flow/task_job_mgr.py

```diff
@@ -1000,16 +1000,17 @@ def _simulation_submit_task_jobs(self, itasks, workflow):
             itask.waiting_on_job_prep = False
             itask.submit_num += 1
             self._set_retry_timers(itask)
-            itask.platform = {'name': 'SIMULATION'}
+            itask.platform['name'] = 'SIMULATION'
             itask.summary['job_runner_name'] = 'SIMULATION'
             itask.summary[self.KEY_EXECUTE_TIME_LIMIT] = (
-                itask.tdef.rtconfig['job']['simulated run length']
+                itask.tdef.rtconfig['simulation']['simulated run length']
             )
             itask.jobs.append(
                 self.get_simulation_job_conf(itask, workflow)
             )
+            itask.state.status = TASK_STATUS_PREPARING
```

Comment:
Otherwise it sticks in the waiting state on retry.

Reply:
This wasn't needed before right?

This gets done somewhere else.

Reply:
Before what? No, resubmit was broken!

Reply:
At the moment (master), I don't think that simulated jobs even go through the preparing state in the first place (which makes sense), we send them straight to submitted:

https://github.com/cylc/cylc-flow/blob/63ceffb21badc2d634a43241ae9a4da75e24189d/cylc/flow/task_job_mgr.py#L997-L1013

I don't think we should change this.

Reply:
At the moment on master the line

```python
            self.task_events_mgr.process_message(
                itask, INFO, TASK_OUTPUT_SUBMITTED
            )
```
just below this chunk converts a task in the waiting state to a task in the running state on the first pass, but doesn't on retry - so that sim mode hangs in waiting state. I'll find out more about why and get back to you.

Reply:
Closing this after in person convo with Oliver - taking a different approach to the resubmit problem.